lib/Interpreter/SQL/MySQLAggregateOptimizer.php

```diff
@@ -78,18 +78,22 @@ public function __construct(Interpreter\Interpreter $interpreter) {
 	private function validateAggregationUsage() {
 		if ($this->aggValid === null) {
 			$this->aggValid = false;
+			$levels = Util\Aggregation::getAggregationLevels();
+			$level = $this->groupBy ? Util\Aggregation::getAggregationLevelTypeValue($this->groupBy) : -1;
```

Comment:
where does the initial level come from if not using `groupBy`?

Reply:
it's caculated by calculateSpeedupGrouping in lib/Interpreter/Interpreter.php